File: Samples/VBHelloWorld/VBGrains/Grain1.vb
Imports Orleans

''' <summary>
''' Orleans grain implementation class Grain1
''' </summary>
Public Class Grain1
    Inherits Grain
    Implements Interfaces.IHello

    Public Function SayHello(greeting As String) As Task(Of String) Implements Interfaces.IHello.SayHello
        Return Task.FromResult("Hello, this is Visual Basic speaking!")
    End Function

End Class

File: Samples/1.x/FSharpHelloWorld/Grains/Grain1.fs
namespace Grains

open System
open System.Threading.Tasks
open Orleans

open FSharpWorldInterfaces

type Grain1() = 
    inherit Orleans.Grain()

    interface IHello with

        override this.SayHello(greeting:string) =
            Task.FromResult<string>("This comes from F#!")
